spring-rabbit/src/test/java/org/springframework/amqp/rabbit/annotation/AbstractRabbitAnnotationDrivenTests.java

```diff
@@ -260,6 +266,50 @@ public void defaultHandle(@Validated String msg) {
 		}
 	}
 
+	/**
+	 * Test for {@link RabbitListenerRepeatableBean} and {@link RabbitListenersBean} that validates that the
+	 * {@code @RabbitListener} annotation is repeatable and generate one specific container per annotation.
+	 */
+	public void testRabbitListenerRepeatable(ApplicationContext context) {
+		RabbitListenerContainerTestFactory simpleFactory =
+				context.getBean("rabbitListenerContainerFactory", RabbitListenerContainerTestFactory.class);
+		assertEquals(2, simpleFactory.getListenerContainers().size());
+
+		MethodRabbitListenerEndpoint first = (MethodRabbitListenerEndpoint)
+				simpleFactory.getListenerContainer("first").getEndpoint();
+		assertEquals("first", first.getId());
+		assertEquals("myQueue", first.getQueueNames().iterator().next());
+
+		MethodRabbitListenerEndpoint second = (MethodRabbitListenerEndpoint)
+				simpleFactory.getListenerContainer("second").getEndpoint();
+		assertEquals("second", second.getId());
+		assertEquals("anotherQueue", second.getQueueNames().iterator().next());
+	}
+
+	@Component
+	static class RabbitListenerRepeatableBean {
+
+		@RabbitListener(id = "first", queues = "myQueue")
+		@RabbitListener(id = "second", queues = "anotherQueue")
+		public void repeatableHandle(String msg) {
```

Comment:
Thanks, that's actually a great suggestion - we recently added a new project `spring-rabbit-test` (adds a test harness to help users with testing `@RabbitListener`). I had to back-off lambda usage and [convert to anonymous classes](https://github.com/spring-projects/spring-amqp/commit/8afa05fd6c262047dca1cea131e69979b44f725a) in a test case there for the same reason.

Perhaps we can move this test there and somehow exclude it from the jdk7 compat build.